src/Maths/Silk.NET.Intrinsics/Common/WorkUnit256.cs

```diff
@@ -0,0 +1,10 @@
+using System.Runtime.Intrinsics;
+
+namespace Silk.NET.Intrinsics
+{
+    internal struct WorkUnit256<T> where T:unmanaged
+    {
+        public WorkUnitFlags Flags { get; set; }
+        public Vector256<T> Vector { get; set; }
```

Comment:
This `WorkUnit` is going to be 64-bytes, seems like a lot of wasted space...

Reply:
Possibly, however typically short-lived.

Reply:
What's the purpose of `WorkUnitFlags`, they look to encode the type and size of the vector, but it isn't clear how its meant to be used in this context?

Reply:
I'm considering removing it as I don't think it's worth having. Originally it was gonna hold which set of instructions to use (i.e. AVX or SSE) so that we can have a static class that redirects maths operations to the correct implementation, however I think we can do that without the flags and probably get better treatment from the JIT too.

Reply:
That sounds reasonable. It would also help with throughput due to reduced memory usage, etc.